### ISSUE TITLE
LaunchServer.sh should cd to the server folder

### ISSUE THREAD

**Issue by dgelessus** [ATLauncher/ATLauncher#125]
The default LaunchServer.sh file (which appears to be shared by all modpacks) currently runs the `java` command in whatever the default directory for the user's shell is. On [some systems](http://forums.atlauncher.com/thread-6523-post-29586.html#pid29586) (specifically OS X) this is the user home directory `~`, and because the LaunchServer script never `cd`s elsewhere, that is where FML and Forge will try to set up the server and load mods from.

This problem could be fixed by adding an extra line above the `java` command:

``` sh
cd "`dirname "$0"`"
```

which will automatically `cd` into the folder in which the LaunchServer script is located before starting the server. I can't think of any situation where this would cause a problem, and if someone wants their mods folder located elsewhere they are probably also smart enough to use a text editor.

Edit: fixed command to work correctly with paths with spaces in them.

**Comment by flaw600** [comment]
So just to confirm, you want the mods to be installed in the same parent directory? That is, the directory where ATLauncher is placed? Sorry, it’s rather late atm and I’m confused as what you mean.

**Comment by dgelessus** [comment]
No, I'm not talking about installation, this is about the default LaunchServer shell script that the launcher creates with every server. Minecraft Forge assumes that the shell's current working directory is the server folder, which isn't true on OS X, where all shells default to starting in the user home (`~`). This `cd` command changes the current directory to the server folder and fixes that problem.

(I'm not sure if this is the correct place to report this, because the file in question is somewhere on the ATL CDN, not in the source code. I couldn't think of any better place to post this though.)

**Comment by flaw600** [comment]
Ah, ok. Is it currently doing this or has it been fixed since the issue was opened?

**Comment by flaw600** [comment]
@dgelessus ?

**Comment by RyanTheAllmighty** [comment]
The LaunchScripts really need to be assessed and updated as they were written well over 2, maybe 3 years ago now.

For reference the current scripts are:

https://download.nodecdn.net/containers/atl/launcher/libraries/LaunchServer.bat
https://download.nodecdn.net/containers/atl/launcher/libraries/LaunchServer.sh

**Comment by dgelessus** [comment]
@RyanTheAllmighty

It would be useful to also add a `java -version` command somewhere in both scripts, that way you can see what the default Java is.

For the `bat` script it would be nice if the script could check the bittiness of Java instead of the OS - it's a really common issue that the default Java is 32-bit even though the system is 64-bit. Not sure if and how that would be possible though.

The `sh` script still needs the fixes described above. Unfortunately that doesn't fix [the entire situation on OS X](https://forums.atlauncher.com/thread-8631.html), where `java` sometimes refers to Java 6 even if a newer version is installed.

**Comment by RyanTheAllmighty** [comment]
Yeah I'll do some digging and researching and see what I can come up with. Definitely been something left on the back burner for a while.

**Comment by dgelessus** [comment]
Ping @RyanTheAllmighty

**Comment by dgelessus** [comment]
Somewhat related - it would also be a good idea to replace the `java` command with the quoted path of the Java version that the user has set in the launcher. The system default Java (or whatever else happens to be the first `java` on the PATH) isn't always the best Java version. For example, on Windows the `java` is sometimes a 32-bit version on 64-bit systems, and on OS X the old Java 6 is often first in the PATH.

**Comment by RyanTheAllmighty** [comment]
Closing in favour of #421 to enhance server installs
